entgql/extension.go

```diff
@@ -379,7 +379,12 @@ func (e *Extension) whereType(t *gen.Type) (string, *ast.InputObjectDefinition,
 			}),
 		}))
 	}
-	fields, err := filterFields(t.Fields)
+
+	allFields := t.Fields
+	if t.ID != nil {
+		allFields = append(allFields, t.ID)
+	}
+	fields, err := filterFields(allFields)
```

Comment:
`t.ID` is always != `nil`.
```suggestion
	fields, err := filterFields(append(t.Fields, t.ID))
```